gb-clock/src/clock.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{Debuger, Ticker};
+use gb_bus::Bus;
+use std::marker::PhantomData;
+
+#[derive(Default)]
+/// Ensure that the various process unit execute their instructions in the right order.
+pub struct Clock<B: Bus<u8> + Bus<u16>> {
+    curr_frame_cycle: usize,
+    phantom_bus: PhantomData<B>,
+}
+
+impl<B: Bus<u8> + Bus<u16>> Clock<B> {
+    /// The amount of cycles to execute per frame.
+    pub const CYCLES_PER_FRAME: usize = 17556;
+
+    /// A single clock cycle, during which each [Ticker] will tick 1 or 4 times depending on their [Tick](crate::Tick) type.
+    pub fn cycle(&mut self, adr_bus: &mut B, process_units: &mut Vec<&mut dyn Ticker<B>>) {
+        for ticker in process_units {
+            for _ in 0..ticker.cycle_count().into() {
+                ticker.tick(adr_bus);
+            }
```

Comment:
Why not adding a `cycle` method to `Ticker` that would do this ?

Reply:
At first I wanted to make `Tick` a generic parameter of `Ticker`, but enums in const generics is an unstable feature requiring nightly for now. So instead I've added the cycle_count function that return a `Tick`.

What I have in mind is that the struct implementing `Ticker`should only select between TCycle and MCycle (with `cycle_count`). That's why I've put this code in the clock. I guess I can add a function with a default implementation into `Ticker` but then there could be custom redefinition of the function. What do you think about it ?

Reply:
Done :)